src/app/user-sign-up/login/login.component.ts

```diff
@@ -1,10 +1,31 @@
 import { Component } from '@angular/core';
+import { FormArray, FormControl, FormGroup, Validators } from '@angular/forms';
 
 @Component({
   selector: 'app-login',
   templateUrl: './login.component.html',
   styleUrls: ['./login.component.scss']
 })
 export class LoginComponent {
+  loginForm!: FormGroup;
+
+  ngOnInit(){
+    this.loginForm = new FormGroup({
+        email : new FormControl('', [Validators.required , Validators.email]),
+        password : new FormControl('',[Validators.required])
+    })
+  }
+
+  get email(){
+    return this.loginForm.get('email');
+  }
+  get password(){
+    return this.loginForm.get('password');
+  }
+
+  onSubmit(){
```

Comment:
Missing return type.
Missing comments. check how the function comments can be written and follow it.